src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/PDController.h

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file PDController.h
+ * @authors Giulio Romualdi
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTROLLERS_OPTIMAL_CONTROL_UTILITIES_PD_CONTROLLER_H
+#define BIPEDAL_LOCOMOTION_CONTROLLERS_OPTIMAL_CONTROL_UTILITIES_PD_CONTROLLER_H
+
+// iDynTree
+#include <iDynTree/Core/Rotation.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/VectorFixSize.h>
+
+namespace BipedalLocomotionControllers
+{
+namespace OptimalControlUtilities
+{
+/**
+ * PDController class is a virtual class that represents a generic PD controller.
+ */
+template <class T, class U, class W> class PDController
+{
+protected:
+    T m_feedforward; /**< Desired feedforward. */
+    U m_referenceDerivative; /**< Reference signal derivative. */
+    W m_reference; /**< Reference signal. */
+
+    U m_stateDerivative; /**< Derivative of the state. */
+    W m_state; /**< State. */
+
+    T m_controllerOutput; /**< Controller output. */
+
+    bool m_controllerOutputEvaluated{false}; /**< Is the controller output evaluated? */
+
+    /**
+     * Evaluate the control output.
+     */
+    virtual void evaluateControl() = 0;
+
+public:
+    /**
+     * Get the controller output.
+     * @return controller output.
+     */
+    const T& getControllerOutput();
+
+    /**
+     * Set the desired trajectory.
+     * @param feedforward
+     * @param referenceDerivative derivative of the reference
+     * @param reference reference signal
+     */
+    void
+    setDesiredTrajectory(const T& feedforward, const U& referenceDerivative, const W& reference);
```

Comment:
``setDesiredPoint``? ``setDesiredValues``? A trajectory assumes that you provide values for different time instants.

Reply:
`setDesiredValues` is fine